wagtail/core/models.py

```diff
@@ -175,16 +182,265 @@ def get_site_root_paths():
         result = cache.get('wagtail_site_root_paths')
 
         if result is None:
-            result = [
-                (site.id, site.root_page.url_path, site.root_url)
-                for site in Site.objects.select_related('root_page').order_by(
-                    '-root_page__url_path', '-is_default_site', 'hostname')
-            ]
+            result = []
```

Comment:
TODO: add a `.local_site_root` attribute to make it easier to find the localised site root page in a template